composeApp/src/commonMain/kotlin/domain/usecase/GetEventInfo.kt

```diff
@@ -7,6 +7,7 @@ import domain.Event
 import kotlinx.datetime.LocalDate
 
 class GetEventInfo(private val eventsRepository: EventsRepository) {
-    suspend fun invoke(date: LocalDate): ResultType<Event, Failure> =
-        eventsRepository.getEvent(date)
+    suspend fun invoke(date: LocalDate): ResultType<Event, Failure> {
```

Comment:
ResultType is from the data layer and it to used in the domain layer

Reply:
I was thinking we should return a Kotlin Result object instead of this custom Result class, and a custom exception as failure. This custom exception can be handled/converted/mapped to a custom Error Object (or just to a String) in an ErrorMapper later

What do you think?